plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2826,6 +2811,16 @@ private static Domain allValues(Type type, boolean includeNull)
         return Domain.notNull(type);
     }
 
+    private static List<DeltaLakeColumnHandle> getColumnHandles(ConnectorTableMetadata tableMetadata, Predicate<ColumnMetadata> columPredicate)
+    {
+        List<String> partitionColumns = getPartitionedBy(tableMetadata.getProperties());
+        List<ColumnMetadata> columnMetadata = tableMetadata.getColumns();
+        return columnMetadata.stream()
+                .filter(columPredicate)
```

Comment:
`getColumnHandles(ConnectorTableMetadata)` would be a better API. Let's make this method do one thing and leave filtering for the callers.

they can easily do that after the fact

```
List<DeltaLakeColumnHandle> columns = getColumnHandles(tableMetadata).stream()
                    .filter(column -> column.getColumnType() != SYNTHESIZED)
                    .collect(toImmutableList());
```